src/config.py

```diff
@@ -0,0 +1,387 @@
+    entropy
+        A random string generated for each project, useful for generating hash keys.
+
+    These properties are only available at runtime:
+
+    branch
+        The Git branch name.
+    environment
+        The environment ID (usually the Git branch plus a hash).
+    documentRoot
+        The absolute path to the web root of the applicatino.
+    smtpHost
+        The hostname of the Platform.sh defauilt SMTP server (an empty string
+        if emails are disabled on the environment.
+
+    """
+
+    # Local index of the variables that can be accessed as direct properties (build and runtime).
+
+    # The key is the property that will be read. The value is the environment variables, minus
+    # prefix, that contains the value to look up.
+
+    directVariables = {'project': 'PROJECT', 'appDir': 'APP_DIR', 'applicationName': 'APPLICATION_NAME',
+                       'treeID': 'TREE_ID', 'entropy': 'PROJECT_ENTROPY'}
+
+    # Local index of the variabels that can be accessed as direct properties (runtime only).
+
+    # The key is the property that will be read. THe value is the environment variables, minus
+    # prefix, that contains the value to look up.
+
+    directVariablesRuntime = {'branch': 'BRANCH', 'environment': 'ENVIRONMENT', 'documentRoot': 'DOCUMENT_ROOT',
+                              'smtpHost': 'SMTP_HOST'}
+
+    # A local copy of all environment variables as of when the object was initialized
+
+    environmentVariables = []
+
+    # The vendor prefix for all environment variables we care about.
+
+    envPrefix = ''
+
+    # The routes definition array.
+
+    # Only available at runtime.
+
+    routesDef = []
+
+    # The relationships definition array.
+
+    # Only available at runtime.
+
+    relationshipsDef = []
+
+    # The variables definition array.
+
+    # Available in both build and runtime, although possibly with different values.
+
+    variablesDef = []
+
+    # The application definition array.
+
+    # This is, approximately, the .platform.app.yaml file in nested array form.
+
+    applicationDef = []
+
+    def __init__(self, environment_variables=None, env_prefix='PLATFORM_'):
+        """Constructs a ConfigReader object.
+
+        :param environment_variables: array|None
+            The environment variables to read. Defaults to the current environment.
+        :param env_prefix: string|None
+            The prefix for environment variables. Defaults to 'PLATFORM_'.
+
+        """
+
+        self.environmentVariables = os.environ if environment_variables is None else environment_variables
+        self.envPrefix = env_prefix
+
+        if self.is_valid_platform():
+
+            if not self.in_build() and self.get_value('ROUTES'):
+
+                routes = self.get_value('ROUTES')
+                self.routesDef = self.decode(routes)
+
+            if not self.in_build() and self.get_value('RELATIONSHIPS'):
+
+                relationships = self.get_value('RELATIONSHIPS')
```

Comment:
This is probably more readable without the blank line between the if and its body block.